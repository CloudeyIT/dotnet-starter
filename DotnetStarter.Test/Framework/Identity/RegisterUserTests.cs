using System;
using System.Threading.Tasks;
using FluentAssertions;
using GraphQL;
using GraphQL.Client.Abstractions;
using NUnit.Framework;
using static DotnetStarter.Core.Framework.Identity.Mutations.RegisterUserMutation;

namespace DotnetStarter.Test.Framework.Identity;

[TestFixture]
public class RegisterUserTests : IntegrationFixture
{
    [Test]
    public async Task Can_Register_User_As_Unauthenticated_User ()
    {
        var request = new GraphQLRequest
        {
            Query = @"
                mutation register($input: RegisterUserInput!) {
                    registerUser(input: $input) {
                        userId
                    }
                }
            ",
            Variables = new
            {
                Input = new RegisterUserInput("test@example.com", "Test", "User", RandomPassword),
            },
        };

        var client = GraphQlClient;

        var response = await client.SendMutationAsync(request, () => new { RegisterUser = new { UserId = Guid.Empty } });

        response.Data.RegisterUser.UserId.Should().NotBeEmpty();
        
        var user = await Database.Users.FindAsync(response.Data.RegisterUser.UserId);
        user.Should().NotBeNull();
        user!.Email.Should().Be("test@example.com");
        user.FirstName.Should().Be("Test");
        user.LastName.Should().Be("User");
    }
}