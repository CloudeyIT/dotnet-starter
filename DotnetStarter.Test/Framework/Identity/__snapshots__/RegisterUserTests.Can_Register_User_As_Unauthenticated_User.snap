{
  "registerUser": {
    "email": "test@example.com",
    "firstName": "Test",
    "lastName": "User",
    "fullName": "Test User"
  }
}
